### PR TITLE
chore(tracing): surface rpc.service in grpc contrib

### PR DESCRIPTION
**Pull request by jbertran**
## Motivation

For upcoming computation of peer service, we want to have the `rpc.service` precursor tag available in GRPC client spans. This tag is the combination of the GRPC package name and the GRPC service name (for example, when emitting a span for a call to the full path `/myPackage.myService/myMethod`, the value of `rpc.service` is `myPackage.myService`).
The tag name is defined in the separate effort for consistent span tags across tracers.

## Change

Surface `rpc.service` in GRPC contrib. The package name may be missing in some cases, and in these cases the value of this tag is _only_ the GRPC service name.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: tests/snapshots/tests.contrib.grpc.test_grpc.test_method_service.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
       "grpc.status.code": "StatusCode.OK",
       "language": "python",
       "network.destination.port": "58143",
+      "rpc.service": "Servicer",
```
Comment (ZStriker19) [inline tests/snapshots/tests.contrib.grpc.test_grpc.test_method_service.json:22]: could you explain Servicer vs pkg.Servicer and how we gain from these distinct values?

Comment (jbertran) [reply tests/snapshots/tests.contrib.grpc.test_grpc.test_method_service.json:22]: This test looks at the trace produced by 2 rpc methods calls: `/Servicer/Handler/` and `/pkg.Servicer/Handler/`.

The span modified here relates to the first call, so we don't have package information, which is expected - there's no duplication of purpose here.

Re: `grpc.method.service` versus `rpc.service`, the purpose of `rpc.service` is to contain package (if available) _and_ service, such that the tag provides a relevant unit of service that can be used by the service map. The end goal is for the new tag to be a precursor of the upcoming `peer.service` tag, which will become the new source of truth for the service map.

**Review by tabgok — APPROVED** [review]

File: tests/snapshots/tests.contrib.grpc.test_grpc.test_method_service.json (hunk 1 above)
Comment (ZStriker19) [reply tests/snapshots/tests.contrib.grpc.test_grpc.test_method_service.json:22]: Ah got it, took a slight mental shift. Thanks for the explanation!

File: releasenotes/notes/report-rpc.service-tag-e6363657f2ccbd5d.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
+#instructions:
```
Comment (ZStriker19) [inline releasenotes/notes/report-rpc.service-tag-e6363657f2ccbd5d.yaml:2]: Let's delete all the commented code here

**Review by tabgok — APPROVED** [review]

File: releasenotes/notes/report-rpc.service-tag-e6363657f2ccbd5d.yaml (hunk 2 above)
Comment (jbertran) [reply releasenotes/notes/report-rpc.service-tag-e6363657f2ccbd5d.yaml]: Done

**Review by tabgok — APPROVED** [review]

**Pull request merged**